federatedscope/vertical/caesar_v_fl/worker/vertical_client.py

```diff
@@ -0,0 +1,367 @@
+                        each for each in self.comm_manager.neighbors
+                        if each != self.server_id
+                    ],
+                    state=self.state,
+                    content=(self.my_public_key, self.others_part_of_my_para)))
+
+    def callback_func_for_public_key_and_para(self, message: Message):
+        self.others_public_key, self.my_part_of_others_para = message.content
+        if self.own_label:
+            self.move_to_the_next_train(None)
+
+    # start training
+    # B sample data
+    # B call encode()
+
+    def move_to_the_next_train(self, message: Message):
+        index, self.input_x, input_y = self.sample_data()
+        self.y = input_y
+        self.batch_index = index
+        self.comm_manager.send(
+            Message(msg_type='sample_data',
+                    sender=self.ID,
+                    receiver=[
+                        each for each in self.comm_manager.neighbors
+                        if each != self.server_id
+                    ],
+                    state=self.state,
+                    content=self.batch_index))
+        self.encode()
+
+    # A sample the same data as B
+    # A also call encode()
+
+    def callback_func_for_sample_data(self, message: Message):
+        index = message.content
+        self.batch_index = index
+        self.input_x = self.sample_data(index=self.batch_index)
+        self.encode()
+
+    # both clients do the following
+    def encode(self):
+        tmp = [
+            self.my_public_key.encrypt(x) for x in self.my_part_of_others_para
+        ]
+        self.comm_manager.send(
+            Message(msg_type='encrypted',
+                    sender=self.ID,
+                    receiver=[
+                        each for each in self.comm_manager.neighbors
+                        if each != self.server_id
+                    ],
+                    state=self.state,
+                    content=tmp))
+
+    def callback_func_for_encrypted(self, message: Message):
+        tmp = message.content
+        self.my_part_of_my_z = np.matmul(self.input_x, self.my_part_of_my_para)
+
+        tmp1 = np.matmul(self.input_x, tmp)
+        self.my_part_of_others_part_of_my_z, tmp = \
+            self.ss.secret_split_for_piece_of_ss(tmp1)
+
+        self.comm_manager.send(
+            Message(msg_type='decrypt',
+                    sender=self.ID,
+                    receiver=[
+                        each for each in self.comm_manager.neighbors
+                        if each != self.server_id
+                    ],
+                    state=self.state,
+                    content=tmp))
+
+    def callback_func_for_decrypt(self, message: Message):
+        tmp1 = message.content
+        self.my_part_of_my_part_of_others_z = [
+            self.my_private_key.decrypt(x) for x in tmp1
+        ]
+        if not self.own_label:
+            self.a_computes()
+
+    # A computes <z>_1 = <z_a>_1 + <<z_a>_2>_1 + <<z_b>_1>_1,
```

Comment:
We can use `client_with_label`/`client_without_label` rather than `A`/ `B` to make this more readable.

Reply:
ok, I'll fix it